client/handle_http.go

```diff
@@ -2256,6 +2243,82 @@ func runPut(request *http.Request, responseChan chan<- *http.Response, errorChan
 
 }
 
+// This function queries the director with a PROPFIND to attempt to get the 'dirListHost'. If a propfind is not allowed on the director
+// We fall back to the deprecated dirlisthost in the namespace
+func getDirListHost(ctx context.Context, remoteObjectUrl *url.URL, namespace namespaces.Namespace, directorUrl string) (dirListHost *url.URL, err error) {
```

Comment:
I think we're trying to get rid of references to `dirListHost`. Wherever you can, let's swap it out for `collectionsUrl`.

Reply:
Good point, can do!